### DISCUSSION TITLE
DB2 Issue while Upgrading to DBB 2.0.1

### DISCUSSION THREAD

**Discussion by Todd0513** [IBM/dbb-zappbuild#528] [General]
I'm getting a -803 on the metadataStore.copyCollection when trying to clone my master branch to my dev branch. Has anyone encountered this and if so, what solution did you find that worked?
com.ibm.db2.jcc.am.SqlIntegrityConstraintViolationException: DB2 SQL Error: SQLCODE=-803, SQLSTATE=23505,SQLERRMC=PCPERMX1;00000043BF, Driver=4.22.37
Caught: groovy.lang.MissingMethodException: No signature of method: build.finalizeBuildProcess() is applicable for argument types: (LinkedHashMap, Integer)

**Comment by dennis-behm** [comment]
Hi @Todd0513 , 

this almost sounds like that you have not applied the database schema changes, that are mandatory for 2.0.1.
Can you please verify, if you have applied those? In the upgrade script from 2.0.0 to 2.0.1 there some constraints are dropped. 
You find them at at the DBB toolkit installation in the: `/usr/lpp/dbb/v2r0/ddl/upgrade_DB2_2.0.1.sql`

Link to the documentation: https://www.ibm.com/docs/en/dbb/2.0?topic=dbb-upgrading-db2-db2-zos-database

Thanks

**Reply by Todd0513** [reply]
Thanks Dennis! Checking on this now.

**Reply by Todd0513** [reply]
Hello Dennis! That was it! A query was missed during our test setup. So I'm past that issue now but now getting a BuildException error, BGZTK01443, user is not authorized to delete collection logical file. I have confirmed that for the DB2 tables, each one has the right accesses, but this seems to be with the MetadataStore collection. How can I see what credentials it is attempting to use for it? Thank you!

**Reply by Todd0513** [reply]
I did find this article that looks like it has my answer: Migrating to IBM Dependency Based Build 2.0 (Part 3)

**Reply by dennis-behm** [reply]
If you migrate from DBB 1.x to DBB 2.0 you need to consider the change in the ownership of the DBB Metadatastore objects. Previously it was most likely a technical user registered in LDAP, now in DBB 2.0 it's a RACF user that will own them. This requires to update the ownership. 

Please see also the [DBB documentation about upgrade](https://www.ibm.com/docs/en/dbb/2.0?topic=overview-upgrading-dbb-20#dbb-db2-database). But also [our blog series](https://www.ibm.com/support/pages/node/6983218) will provide valuable pointers to migrate (hopefully) seamlessly.

**Reply by dennis-behm** [reply]
@Todd0513 - Is your msg `BGZTK0144E`. Are you on DBB 2.0.2.1? 

Can you double-check the owner of the collections please? Has that changed?

**Reply by Todd0513** [reply]
Thanks Dennis! I will get that submitted. Any thoughts on what is causing it?

**Reply by dennis-behm** [reply]
@Todd0513 I ran into a similar issue recently:

I made the mistake and forgot that zAppBuild clones DBB collections for feature branches. The cloning then copies the ownership of collections.

In my case the **owner** of the collection of the mainBuildBranch let's say `MortgageApplication-main` was my personal user (I ran some tests from the cli), while my build was executed under the pipeline orchestrator user (Jenkins) which is then not allowed to update the copied collection metadata for the feature branch because of the mismatch of the ownership.

Please double check the owners of your collections and the user that is running the pipeline build.

**Reply by Todd0513** [reply]
Good to know! To check these, this is the data stored in the DB2 tables for the collections and the permissions tables?

**Reply by dennis-behm** [reply]
Yes, It's in the DBBCOLLECTION table. Alternatively, you can also use the [DBB CLI ](https://www.ibm.com/docs/en/dbb/2.0?topic=interface-legacy-commands)to inspect the owner. I recommend to stay with the legacy cmds  .

```
dbb collection view MortgageApplication-main-outputs --type db2 --db2-config /var/dbb/config/db2Connection_Jenkins.conf --password-file /var/dbb/config/db2-pwd-file.xml --user DBEHM
```

**Reply by Todd0513** [reply]
Thanks Dennis! This is VERY helpful!

**Comment by Todd0513** [comment]
@dennis-behm Hello Dennis! I have another question on the DBB 2.0 upgrade...in my test project, I removed a source code but left the related copybooks because I still have other source using those copybooks. But when I change one of those copybooks to test the impact, it still attempts to pull in the source I removed.
To remove it, I simply deleted it from the project. Is there another process I should have done to have "properly" updated the impact data? And now that I have removed it, is there a way to clean-up the impact data?
Thanks!

**Reply by dennis-behm** [reply]
@Todd0513  - would you mind to open another discussion thread for this, please? I'll respond to this one then in detail.

In short: computing the relative path of a _deleted_ file (dependency information are stored in the DBB Metadatastore per repository file) is tricky, because we cannot test if the file exists. zAppBuild already supports multiple repository layouts, and I assume the default settings which is assumed for the deletion case, does not match yours.

**Reply by Todd0513** [reply]
Yes, I'll open a new thread. Thanks Dennis!
